### PR TITLE
fix variable check in xcode_backend.sh

### LINKED ISSUE #24343: No code signature found - Add Flutter to existing apps, iOS real device issue, Xcode Version 10.1
After doing everything listed here:
https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps

I get:

screen shot 2018-11-14 at 12 48 13 pm

when trying to run app on any real device. 

**On simulator, it all works as expected.**

**On Android side, everything goes smoothly, also!**

## Steps to Reproduce

1. Create simple iOS project(Swift) in the same folder where the flutter module is
2. Initialize CocoaPods and change Podfile to

```
# Uncomment the next line to define a global platform for your project
platform :ios, '10.0'
# use_frameworks!

flutter_application_path = '../flutter_comment_wall/'
  eval(File.read(File.join(flutter_application_path, '.ios', 'Flutter', 'podhelper.rb')), binding)

target 'hostiosapp' do

end
```
3. pod install
screen shot 2018-11-14 at 1 13 26 pm

4. add those 
```
"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build
"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" embed
```
to BuildPhases Run Script
5. Make simple button and onClick listener to open FlutterViewController
6. Build successful, after I set:
screen shot 2018-11-14 at 1 05 51 pm
without this, i get a linking error.
```
ld: '/Users/jelenalecic/Library/Developer/Xcode/DerivedData/hostiosapp-ahpyodppdcdpvkbrkpickvrsdqea/Build/Products/Debug-iphoneos/FlutterPluginRegistrant/libFlutterPluginRegistrant.a(GeneratedPluginRegistrant.o)' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target. file '/Users/jelenalecic/Library/Developer/Xcode/DerivedData/hostiosapp-ahpyodppdcdpvkbrkpickvrsdqea/Build/Products/Debug-iphoneos/FlutterPluginRegistrant/libFlutterPluginRegistrant.a' for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
7. App installation failed, No code signature found

Signing is set to automatically

screen shot 2018-11-14 at 1 17 40 pm

Device that I'm currently using is an iphone 8, iOS 12.1

flutter doctor:

```
[✓] Flutter (Channel beta, v0.11.3, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.11.3 at /Users/jelenalecic/projects/flutter
    • Framework revision 72bf075e8d (4 days ago), 2018-11-09 20:36:17 -0800
    • Engine revision 5646e86a6f
    • Dart version 2.1.0 (build 2.1.0-dev.9.3 9c07fb64c4)

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at /Users/jelenalecic/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 30.0.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] Connected device (3 available)
    • SM C7010            • d571e02b                                 • android-arm64 • Android 8.0.0 (API 26)
    • Jelena’s iPhone (2) • ac1e7e4b9bc055e0c4ff05c3bcbe64e189b6480e • ios           • iOS 12.1
    • iPhone 7            • 9DC8517E-E336-4BF8-A2C6-9A65DC0DB063     • ios           • iOS 12.1 (simulator)

• No issues found!

```

I believe it must be some binding issue with podhelper.rb, 'cos i have to  switch off manually Enable Bitcode, to be able to build the app.

I have tried everything I could think of. Please help :)
Thanks!

@dnfield

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/24343

The current test in there is wrong.

This also removes some trailing whitespace because my editor is now configured to do that automatically.

### CODE REVIEW

File: packages/flutter_tools/bin/xcode_backend.sh [hunk 1]
```diff
@@ -360,14 +360,14 @@ EmbedFlutterFrameworks() {
...
+  # Remove it first since Xcode might be trying to hold some of these files - this way we're
   # sure to get a clean copy.
   RunCommand rm -rf -- "${xcode_frameworks_dir}/Flutter.framework"
   RunCommand cp -Rv -- "${flutter_ios_engine_folder}/Flutter.framework" "${xcode_frameworks_dir}/"
-  
+
   # Sign the binaries we moved.
   local identity="${EXPANDED_CODE_SIGN_IDENTITY_NAME:-$CODE_SIGN_IDENTITY}"
-  if (${identity} != ""); then
+  if [[ ! -z "$identity" ]]; then
```
Comment (gspencergoog) [inline packages/flutter_tools/bin/xcode_backend.sh]: Why not use `-n`?

**Review by gspencergoog — CHANGES_REQUESTED** [review]

**Review by goderbauer — APPROVED** [review]
LGTM

Any chance we can write a tests that would have caught this?

File: packages/flutter_tools/bin/xcode_backend.sh (hunk 1 above)
Comment (dnfield) [reply packages/flutter_tools/bin/xcode_backend.sh]: Done

**Review by gspencergoog — APPROVED** [review]

**Comment by dnfield** [comment]
@goderbauer - I have some tests for this script, but it's hard to catch some of these that are more deeply nested without a full on integration test.  I ended up reproducing this while working on an integration test that would catch it in the future.

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

**Pull request merged**